### PR TITLE
TTT: Corpses get skin&bodygroups

### PR DESCRIPTION
**Pull request by aceradryd**
TTT: Corpses get the skin and bodygroups from the dead Player.

### CODE REVIEW

File: garrysmod/gamemodes/terrortown/gamemode/corpse.lua [hunk 1]
```diff
@@ -386,6 +386,11 @@ function CORPSE.Create(ply, attacker, dmginfo)
 
    rag:SetPos(ply:GetPos())
    rag:SetModel(ply:GetModel())
+   rag:SetSkin(ply:GetSkin())
+   bodyGroupTable = ply:GetBodyGroups()
+   for id in pairs(bodyGroupTable) do
```
Comment (svdm) [inline garrysmod/gamemodes/terrortown/gamemode/corpse.lua]: You're creating a `bodyGroupTable` global var here. Just make it `for id in pairs(ply:GetBodyGroups()) do`

**Comment by aceradryd** [comment]
I have changed the code.

Yes, you can set your skin/bodygroup for example with the "Enhanced PlayerModel Selector" and then the corpse should have the skin/bodygroup also for example for the "Advanced Disguiser".

**Comment by Kefta** [comment]
Shouldn't you be using the "id" key in the value table instead of using the key? Don't know if bodygroups work sequentially or not.

**Comment by aceradryd** [comment]
I dont know, what do you mean, Kefta?

**Comment by Kefta** [comment]
GetBodyGroups returns a table with the bodygroup id key in it. Currently, you are just using the table index. This should work so long as the bodygroups are sequential  (1, 2, 3 ,..., k) but I'm not sure if that's guaranteed or not.

**Comment by aceradryd** [comment]
Okay, i have changed the code for this case. Now it use the key from the table.

**Comment by BlackCetha** [comment]
You should use the dot-notation for accessing string-indexed tables (`table.index`) and add a space after the comma to keep the style consistent with the codebase.

**Comment by aceradryd** [comment]
Changed

**Pull request merged**